### PR TITLE
fix(tier4_simulation_msgs): bring message of dummy object to simulator msgs

### LINKED ISSUE #8704: Move dummy_perception_msgs.Objects message definition to autoware_msgs
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I've agreed with the maintainers that I can plan this task.

### Description

I suggest to move `dummy_perception_msgs.Objects` message declaration in `dummy_perception_publisher` into https://github.com/autowarefoundation/autoware_msgs.
This is related to https://github.com/autowarefoundation/autoware.universe/issues/8695.

### Purpose

We want to remove `common` folder packages dependency to `dummy_perception_publisher` and it seems to depend its message named `dummy_perception_msgs.Objects`.

I think we can move this declaration to autoware_msgs.

### Possible approaches

I think there are some approaches:

1. move `dummy_perception_msgs.Objects` to `autoware_msgs/autoware_perception_msgs`
2. create `autoware_msgs/autoware_simulation_msgs` and put the message into it

Case 1 needs message name change, while case 2 won't need it.
I prefer the latter, but I want to you guys advise.

### Definition of done

Once after message transfer and dependency will be solved.

### PR DESCRIPTION
**Pull request by technolojin**
## Related Links

Issue https://github.com/autowarefoundation/autoware.universe/issues/8704

[TIER IV INTERNAL DISCUSSION](https://star4.slack.com/archives/C4P0NSMB5/p1726016981920039?thread_ts=1724999986.645329&cid=C4P0NSMB5)

## Description
Bringing dummy object message from `dummy_perception_publisher` package.

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [ ] Code is properly formatted
- [ ] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code is properly formatted
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR**: Required to pass before the merge.
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Review by youtalk — APPROVED** [review]
LGTM
See also https://github.com/autowarefoundation/autoware.universe/pull/8828#pullrequestreview-2294807666.

File: tier4_simulation_msgs/CMakeLists.txt [hunk 1]
```diff
@@ -20,7 +20,15 @@ rosidl_generate_interfaces(${PROJECT_NAME}
   msg/SimulationEvents.msg
   msg/UserDefinedValue.msg
   msg/UserDefinedValueType.msg
-  DEPENDENCIES builtin_interfaces std_msgs)
+  msg/DummyObject.msg
+  msg/DymmyObjectInitialState.msg
```
Comment (yukkysaito) [inline tier4_simulation_msgs/CMakeLists.txt]:
```suggestion
  msg/DummyObjectInitialState.msg
```

File: tier4_simulation_msgs/msg/DummyObject.msg [hunk 2]
```diff
@@ -0,0 +1,13 @@
+std_msgs/Header header
+unique_identifier_msgs/UUID id
+tier4_simulation_msgs/DymmyObjectInitialState initial_state
```
Comment (yukkysaito) [inline tier4_simulation_msgs/msg/DummyObject.msg]:
```suggestion
tier4_simulation_msgs/DummyObjectInitialState initial_state
```

File: tier4_simulation_msgs/msg/DymmyObjectInitialState.msg
Comment (yukkysaito) [inline tier4_simulation_msgs/msg/DymmyObjectInitialState.msg:1]: Pls fix wrong file name

Comment (technolojin) [reply tier4_simulation_msgs/msg/DymmyObjectInitialState.msg:1]: fixed https://github.com/tier4/tier4_autoware_msgs/pull/146/commits/6d15270454b091a8298717a168679e38229198b7

**Review by yukkysaito — APPROVED** [review]

**Comment by technolojin** [comment]
@yukkysaito I am not authorized to merge this PR. Can you proceed it?

**Comment by yukkysaito** [comment]
@technolojin Sure. Please wait a moment while I check if there are any impacts from the interface changes.

Please provide the steps to perform the operation check. Should I merge the following PR?
- https://github.com/tier4/tier4_autoware_msgs/pull/146
- https://github.com/autowarefoundation/autoware.universe/pull/8828

**Comment by technolojin** [comment]
@yukkysaito 
I described the steps in the parent issue https://github.com/autowarefoundation/autoware.universe/issues/8704#issuecomment-2342585466

**Pull request merged**